src/Platforms/Gcc/UtestPlatform.cpp

```diff
@@ -232,12 +232,18 @@ void (*PlatformSpecificFree)(void* memory) = free;
 void* (*PlatformSpecificMemCpy)(void*, const void*, size_t) = memcpy;
 void* (*PlatformSpecificMemset)(void*, int, size_t) = memset;
 
-/* MinGw GCC 5.3.0 isnan / isinf macros are not working properly, causing a conversion
+/* GCC 4.9+ isnan / isinf macros are not working properly, causing a conversion
  * warning / error
  */
-#if defined(__GNUC__) && __GNUC__ >= 5
+#if ( defined(__GNUC_PREREQ) && __GNUC_PREREQ(4, 9) )
```

Comment:
Maybe check whether it's defined on a separate line? The preprocessor is a bit simple-minded ;-)